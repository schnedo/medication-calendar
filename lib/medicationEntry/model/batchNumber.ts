export default interface BatchNumber {
  readonly number: number;
}

const prefix = "Ch.-B.";
export function format(batchNumber: BatchNumber): string {
  return `${prefix} ${batchNumber.number}`;
}
