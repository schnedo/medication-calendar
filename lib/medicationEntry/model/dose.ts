export default interface Dose {
  readonly amount: number;
}

const unit = "ml";

export function format(dose: Dose): string {
  return `${dose.amount} ${unit}`;
}
