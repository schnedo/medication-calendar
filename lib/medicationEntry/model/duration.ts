export default interface Duration {
  minutes: number;
  hours: number;
}

function formatPart(number: number | undefined) {
  if (number === undefined) {
    return "00";
  }
  const raw = number.toString();
  return raw.length === 1 ? `0${raw}` : raw;
}

export function format(duration: Duration): string {
  return `${formatPart(duration.hours)}:${formatPart(duration.minutes)}`;
}
