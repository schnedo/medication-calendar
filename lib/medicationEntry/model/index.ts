export { default as BatchNumber } from "./batchNumber";
export { default as Dose } from "./dose";
export { default as Duration } from "./Duration";
export { default as Medicament } from "./medicament";
export { default as Medication } from "./medication";
export { default as MedicationEntry } from "./medicationEntry";

export * as Repository from "./repository";
