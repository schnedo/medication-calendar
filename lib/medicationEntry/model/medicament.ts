import BatchNumber from "./batchNumber";

export default interface Medicament {
  readonly name: string;
  readonly batchNumber: BatchNumber;
}
