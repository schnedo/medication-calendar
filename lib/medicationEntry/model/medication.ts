import Dose from "./dose";
import Medicament from "./medicament";

export default interface Medication {
  readonly id: string;
  readonly dose: Dose;
  readonly medicament: Medicament;
}
