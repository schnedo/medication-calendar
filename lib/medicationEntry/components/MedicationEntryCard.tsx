import { ReactElement } from "react";
import { Card, CardContent, makeStyles, Typography } from "@material-ui/core";
import { MedicationCard } from "../components";
import { formatDuration, MedicationEntry } from "../model";
import { format } from "date-fns";
import { formatBodyMass } from "../../contact";

const useStyles = makeStyles((theme) => ({
  header: {
    marginRight: theme.spacing(1),
  },
  trailer: {
    marginLeft: theme.spacing(1),
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
  },
  row: {
    display: "flex",
  },
  rowItem: {
    flex: "1",
  },
  medicationsList: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
  },
  medicationsListItem: {
    flex: "1",
    marginBottom: theme.spacing(1),
    "&:lastChild": 0,
  },
}));

export interface MedicationEntryCardProps {
  medicationEntry: MedicationEntry;
}

export default function MedicationEntryCard({
  medicationEntry: { date, medications, comments, duration, bodyMass },
}: MedicationEntryCardProps): ReactElement {
  const {
    medicationsListItem,
    medicationsList,
    row,
    rowItem,
    header,
    trailer,
  } = useStyles();
  return (
    <Card>
      <CardContent className={row}>
        <Typography className={header}>{format(date, "EEEEEE dd")}</Typography>
        <div className={`${medicationsList} ${rowItem}`}>
          {medications.map((medication) => (
            <MedicationCard
              medication={medication}
              className={medicationsListItem}
              key={medication.id}
            />
          ))}
          <Typography>{comments}</Typography>
        </div>
        <div className={trailer}>
          <Typography>{formatDuration(duration)}</Typography>
          <Typography>{formatBodyMass(bodyMass)}</Typography>
        </div>
      </CardContent>
    </Card>
  );
}
