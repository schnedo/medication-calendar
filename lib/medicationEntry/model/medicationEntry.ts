import Medication from "./medication";
import { BodyMass } from "../../contact/model";
import Duration from "./duration";

export default interface MedicationEntry {
  readonly id: string;
  readonly date: Date;
  readonly medications: Medication[];
  readonly duration: Duration;
  readonly comments: string;
  readonly bodyMass: BodyMass;
}
