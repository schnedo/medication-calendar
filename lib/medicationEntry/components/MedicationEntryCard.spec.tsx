import React from "react";
import { render } from "@testing-library/react";
import MedicationEntryCard from "./MedicationEntryCard";
import { MedicationEntry } from "../model";

jest.mock("../components", () => ({
  MedicationCard() {
    return <div>mocked</div>;
  },
}));

describe("MedicationEntryCard", () => {
  it("should render correctly", async () => {
    expect.hasAssertions();

    const medicationEntry: MedicationEntry = {
      bodyMass: {
        amount: 75,
      },
      comments: "comments",
      date: new Date(2020, 1, 11),
      duration: { minutes: 15, hours: 1 },
      id: "1",
      medications: [
        {
          id: "2",
          dose: {
            amount: 200,
          },
          medicament: {
            name: "name",
            batchNumber: {
              number: 12345,
            },
          },
        },
      ],
    };

    const { container } = render(
      <MedicationEntryCard medicationEntry={medicationEntry} />,
    );

    expect(container).toMatchSnapshot();
  });
});
