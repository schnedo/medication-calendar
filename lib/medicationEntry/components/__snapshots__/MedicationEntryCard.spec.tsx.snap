// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationEntryCard should render correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-row-3"
    >
      <p
        class="MuiTypography-root makeStyles-header-1 MuiTypography-body1"
      >
        Tu 11
      </p>
      <div
        class="makeStyles-medicationsList-5 makeStyles-rowItem-4"
      >
        <div>
          mocked
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          comments
        </p>
      </div>
      <div
        class="makeStyles-trailer-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          01:15
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          75 kg
        </p>
      </div>
    </div>
  </div>
</div>
`;
